cashu/wallet/api.py

```diff
@@ -0,0 +1,360 @@
+from pydantic import BaseModel, PositiveInt, Field, SecretStr
+from fastapi import HTTPException, status, APIRouter, Path, Query
+from pathlib import Path
+from operator import itemgetter
+from itertools import groupby
+from datetime import datetime
+from cashu.core.base import Proof
+from cashu.core.helpers import sum_proofs
+from cashu.core.migrations import migrate_databases
+from cashu.wallet import migrations
+from cashu.wallet.crud import (
+    get_lightning_invoices,
+    get_reserved_proofs,
+    get_unused_locks,
+)
+from cashu.wallet.wallet import Wallet as Wallet
+from cashu.wallet.settings import MintSettings, CashuSettings
+
+
+logger = logging.getLogger(__name__)
+
+app = APIRouter(prefix="/v0/wallet")
+
+mint_settings = MintSettings()
+cashu_settings = CashuSettings()
+
+
+async def init_wallet(wallet_c: Wallet):
+    """Performs migrations and loads proofs from db."""
+    await migrate_databases(wallet_c.db, migrations)
+    await wallet_c.load_proofs()
+
+
+class WalletModel(BaseModel):
+    """Wallet model"""
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    wallet_name: str = "default"
+    wallet: Wallet = Wallet(
+        url=mint_settings.mint_url,
+        db=str(Path(cashu_settings.cashu_dir) / wallet_name),
+    )
+    active_wallet: bool
+
+
+wallet = WalletModel(active_wallet=True)
+
+
+@app.on_event("startup")
+async def start_wallet():
+    """Starts wallet on startup."""
+    await init_wallet(wallet.wallet)
+    await wallet.wallet.load_mint()
+    logger.info("Wallet started")
+
+
+@app.on_event("shutdown")
+async def shutdown_tor():
+    """Shutdowns wallet on shutdown."""
+    if wallet.wallet.tor.tor_running:
+        logger.info("Stopping Tor")
+        wallet.wallet.tor.stop_daemon()
+    logger.info("Wallet shutdown")
+
+
+@app.get("/health")
+async def health():
+    """Health check"""
+    return {"status": "OK"}
+
+
+@app.post("/lightning/pay")
+async def pay_lightning_invoice(invoice: str):
+    """Pay a lightning invoice."""
+    if not cashu_settings.lightning:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Lightning is not enabled.",
+        )
+    logger.info(f"Wallet status: {wallet.wallet.status()}")
+    amount, fees = await wallet.wallet.get_pay_amount_with_fees(invoice)
+    if amount <= 0:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Amount must be greater than 0.",
+        )
+    if amount + fees > wallet.wallet.available_balance:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Insufficient funds.",
+        )
+    _, send_proofs = await wallet.wallet.split_to_send(wallet.wallet.proofs, amount)
+    await wallet.wallet.pay_lightning(send_proofs, invoice)
+    logger.info(f"Wallet status: {wallet.wallet.status()}")
+    return {"amount": amount, "fees": fees}
+
+
+@app.get("/lightning/invoice")
```

Comment:
Difference between `lightning/invoice` and `lightning/receive` is unclear. Please stick to exactly the same names as in the CLI, do not invent new names if not necessary.

Reply:
Sorry about inventing new names. 
I added this function so the `invoice` path didn't hang waiting for conf. that the invoice has been paid ie to replace:
```python
            while time.time() < check_until and not paid:
                time.sleep(3)
                try:
                    await wallet.mint(amount, invoice.hash)
                    paid = True
                    print(" Invoice paid.")
                except Exception as e:
                    # TODO: user error codes!
                    if str(e) == "Error: Lightning invoice not paid yet.":
                        print(".", end="", flush=True)
                        continue
```
from the `cli` maybe that's a wrong. I can make it hang.
Else, i can make this a `invoice` "status" path.
What do you think is more suited?